sentry_core.go

```diff
@@ -193,3 +221,37 @@ func (s *SentryCore) Sync() error {
 	s.hub.Flush(30 * time.Second)
 	return nil
 }
+
+func (s *SentryCore) parseFieldsToEvent(event *sentry.Event, data map[string]interface{}) {
+	event.User = s.prepareSentryUser(&data)
+	event.Tags = s.prepareSentryTags(&data)
+	event.Extra = data
+}
+
+func (s *SentryCore) prepareSentryUser(data *map[string]interface{}) sentry.User {
+	return sentry.User{
+		ID:        fmt.Sprintf("%v", pop(data, s.TagMap.UserTags.ID)),
+		IPAddress: fmt.Sprintf("%v", pop(data, s.TagMap.UserTags.IPAddress)),
+		Name:      fmt.Sprintf("%v", pop(data, s.TagMap.UserTags.Name)),
+		Username:  fmt.Sprintf("%v", pop(data, s.TagMap.UserTags.Username)),
+		Email:     fmt.Sprintf("%v", pop(data, s.TagMap.UserTags.Email)),
+		Segment:   fmt.Sprintf("%v", pop(data, s.TagMap.UserTags.Segment)),
+	}
+}
+
+func (s *SentryCore) prepareSentryTags(data *map[string]interface{}) map[string]string {
+	tags := make(map[string]string, 0)
+	for _, tagKey := range s.TagMap.GenericTags {
+		tags[tagKey] = fmt.Sprintf("%v", pop(data, tagKey))
```

Comment:
Is sending empty value to Sentry a desired behavior if field doesn't exist in zap event? Probably it's better to filter them client side even if Sentry will filter them anyway